test/pylib/manager_client.py

```diff
@@ -289,9 +289,20 @@ async def servers_add(self, servers_num: int = 1,
 
     async def remove_node(self, initiator_id: ServerNum, server_id: ServerNum,
                           ignore_dead: List[IPAddress] | List[HostID] = list[IPAddress](),
-                          expected_error: str | None = None) -> None:
+                          expected_error: str | None = None,
+                          wait_removed_dead: bool = True) -> None:
         """Invoke remove node Scylla REST API for a specified server"""
         logger.debug("ManagerClient remove node %s on initiator %s", server_id, initiator_id)
+
+        # If we remove a node, we should wait until other nodes see it as dead
+        # because the removenode operation can be rejected if the node being
+        # removed is considered alive. However, we sometimes do not want to
+        # wait, for example, when we test that removenode fails as expected.
+        # Therefore, we make waiting optional and default.
```

Comment:
@gleb-cloudius heads up, you might want to adjust your tests once this is merged.